src/web/app/pages-admin/admin-notifications-page/notification-edit-form/notification-edit-form.component.spec.ts

```diff
@@ -1,30 +1,69 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
+import moment from 'moment-timezone';
+import { TimezoneService } from '../../../../services/timezone.service';
+import { NotificationTargetUser } from '../../../../types/api-output';
+import { EXAMPLE_NOTIFICATION_EDIT_MODEL } from '../admin-notifications-page-data';
 import { AdminNotificationsPageModule } from '../admin-notifications-page.module';
-
+import { NotificationEditFormModel } from './notification-edit-form-model';
 import { NotificationEditFormComponent } from './notification-edit-form.component';
 
 describe('NotificationEditFormComponent', () => {
   let component: NotificationEditFormComponent;
   let fixture: ComponentFixture<NotificationEditFormComponent>;
+  let timezoneService: TimezoneService;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [
         AdminNotificationsPageModule,
         HttpClientTestingModule,
       ],
+      providers: [
+        TimezoneService,
+      ],
     })
     .compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(NotificationEditFormComponent);
     component = fixture.componentInstance;
+    timezoneService = TestBed.inject(TimezoneService);
+    moment.tz.setDefault('UTC');
+    jest.spyOn(timezoneService, 'guessTimezone').mockReturnValue('Asia/Singapore');
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default view', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('set up with EXAMPLE_NOTIFICATION_EDIT_MODEL', () => {
```

Comment:
Would be good for the description of this and the following testcase to also begin with "should" for consistency with the rest of the testcases.